### DISCUSSION TITLE
Technical flutter js_interop questions

### DISCUSSION THREAD

**Discussion by redDwarf03** [archethic-foundation/wagmi_flutter_web#3] [General]
Why jsArgs.add doesn't work ?
```
  JSArray<JSObject>? _convertArgs(List<dynamic>? args) {
    if (args == null) {
      return null;
    }
    final jsArgs = JSArray<JSObject>();
    for (final arg in args) {
      if (arg is String) {
        jsArgs.add(arg.toJS);
      } else if (arg is int) {
        jsArgs.add(arg.toJS);
      } else if (arg is bool) {
        jsArgs.add(arg.toJS);
      }
    }
    return jsArgs;
  }
```

**Comment by redDwarf03** [comment]
Issue with estimateGas

Error 
```
main.js:179 Error estimateGas: TypeError: Cannot read properties of null (reading 'toString')
    at formatUnits (index-BR7LYkqh.js:39854:13)
    at formatGwei (index-BR7LYkqh.js:39864:10)
    at new EstimateGasExecutionError (index-BR7LYkqh.js:41059:43)
    at getEstimateGasError (index-BR7LYkqh.js:41089:10)
    at estimateGas$1 (index-BR7LYkqh.js:41129:11)
``` 
Because it tries to convert nullable bigint

So i try to fix it with

```
  JSEstimateGasParameters get toJS => JSEstimateGasParameters(
        accessList: accessList?.jsify() as JSArray<JSObject>?,
        account: account?.toJS,
        chainId: chainId?.toJS,
        data: data?.toJS,
        gas: gas?.toJS ?? BigInt.zero.toJS,
        gasPrice: gasPrice?.toJS ?? BigInt.zero.toJS,
        maxFeePerGas: maxFeePerGas?.toJS ?? BigInt.zero.toJS,
        maxPriorityFeePerGas: maxPriorityFeePerGas?.toJS ?? BigInt.zero.toJS,
        nonce: nonce?.toJS,
        to: to?.toJS,
        type: type?.toJS,
        value: value?.toJS,
      );
```

But i have a FeeConflictError error because Cannot specify both a `gasPrice` and a `maxFeePerGas`/`maxPriorityFeePerGas`.
Use `maxFeePerGas`/`maxPriorityFeePerGas` for EIP-1559 compatible networks, and `gasPrice` for others.

**Reply by kaumudpa** [reply]
@redDwarf03 were you able to resolve this? - if yes how & is it implemented?

**Reply by redDwarf03** [reply]
not yet

**Reply by kaumudpa** [reply]
Many important implementations will require passing `gasPrice` or `maxFeePerGas/maxPriorityFeePerGas` like `sentTransaction` 

We have successfully implemented `sendTransaction` only with `gasPrice` which sends a legacy transaction but for EIP-1559 transaction we are facing similar issue where - `TypeError: Cannot read properties of null (reading 'toString')`

Any suggestions?

**Reply by redDwarf03** [reply]
We are on it

**Reply by redDwarf03** [reply]
i have something to share with you in few minutes... to be challenged

**Reply by kaumudpa** [reply]
Ok. Waiting.

**Reply by redDwarf03** [reply]
https://github.com/archethic-foundation/wagmi_flutter_web/pull/13

**Reply by redDwarf03** [reply]
estimate gas merged. Don't hesitate to try it with new examples

**Reply by kaumudpa** [reply]
I have opened a PR which needs some help. Can you guys please look into it?
